diff -ur lua-5.2.0/src/Makefile lua-5.2.0-shared/src/Makefile
--- lua-5.2.0/src/Makefile	2011-09-19 14:45:14.000000000 +0200
+++ lua-5.2.0-shared/src/Makefile	2012-06-12 22:24:16.973276561 +0200
@@ -103,7 +103,7 @@
 generic: $(ALL)
 
 linux:
-	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -ldl -lreadline -lncurses"
+	$(MAKE) $(ALL) SYSCFLAGS="-fPIC -DLUA_USE_LINUX" SYSLIBS="-Wl,-E -ldl -lreadline -lncurses"
 
 macosx:
 	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_MACOSX" SYSLIBS="-lreadline"
